src/modules/builtin/cd.rs

```diff
@@ -20,7 +21,15 @@ impl SyntaxModule<ParserMetadata> for Cd {
 
     fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
         token(meta, "cd")?;
+        let tok = meta.get_current_token();
         syntax(meta, &mut self.value)?;
+        let path_type = self.value.get_type();
+        if path_type != Type::Text {
+            return error!(meta, tok => {
+                message: "Builtin function `cd` can only be used with values of type Text",
+                comment: format!("Given type: {}, expected type: {}", path_type, Type::Text)
+            });
+        }
```

Comment:
So basically this code will print error highlighting just the first token. This will work for `cd 12` but not for some longer expression `cd 12 + 15`. Not so long ago I've added a new API for Expr so that we can easily highlight them.

However I forgot to write a convenience function for it. Can you please replace this piece of code from `src/modules/expression/expr.rs`:
```rs
impl Expr {
    pub fn get_error_message(&self, meta: &mut ParserMetadata) -> Message {
        let begin = meta.get_token_at(self.pos.0);
        let end = meta.get_token_at(self.pos.1);
        let pos = PositionInfo::from_between_tokens(meta, begin, end);
        Message::new_err_at_position(meta, pos)
    }
```
with this:
```rs
impl Expr {
    pub fn get_position(&self, meta: &mut ParserMetadata) -> PositionInfo {
        let begin = meta.get_token_at(self.pos.0);
        let end = meta.get_token_at(self.pos.1);
        PositionInfo::from_between_tokens(meta, begin, end)
    }

    pub fn get_error_message(&self, meta: &mut ParserMetadata) -> Message {
        let pos = self.get_position(meta);
        Message::new_err_at_position(meta, pos)
    }
```

This adds new function that will be useful for this case that you have. Let's see how it can be used:
```suggestion
        syntax(meta, &mut self.value)?;
        let path_type = self.value.get_type();
        if path_type != Type::Text {
            return error_pos!(meta, self.expr.get_position(meta) => {
                message: "Builtin function `cd` can only be used with values of type Text",
                comment: format!("Given type: {}, expected type: {}", path_type, Type::Text)
            });
        }
```